lister.cc

```diff
@@ -75,3 +77,44 @@ future<> lister::rmdir(fs::path dir) {
         return remove_file(dir.native());
     });
 }
+
+future<std::optional<directory_entry>> queue_lister::get() {
+    if (!_done) {
+        auto f = co_await open_checked_directory(general_disk_error_handler, _dir.native());
+        auto walker = [this] (fs::path dir, directory_entry de) {
+            return _q.push_eventually(std::make_optional<directory_entry>(std::move(de))).handle_exception([] (std::exception_ptr) {
+                // ignore exception, queue was aborted
+            });
+        };
+        assert(!_lister);
+        _lister = std::make_unique<lister>(std::move(f), _type, std::move(walker), _filter, _dir, _do_show_hidden);
+        auto done_fut = _lister->done().then_wrapped([this] (future<> f) {
```

Comment:
nitpick: I think you can just do `_done` = ... here. I think (but didn't test) that when _done is optional<future> you can assign to it a future and that's it - I think you don't need to call `std::make_optional` as you did below.

Reply:
I'll try that...

Reply:
Fixed in v3